File: 0027_easy_hot_remove-element/solution.java
class Solution {
    public int removeElement(int[] nums, int val) {
        if(nums == null || nums.length == 0) {
            return 0;
        }

        int n = nums.length;
        int i = n - 1;

        while (i >= 0 && nums[i] == val) {
            i--;
        }

        for (int j = i-1; j > -1; j--) {
            if (nums[j] == val) {
                int temp = nums[i];
                nums[i] = nums[j];
                nums[j] = temp;
                i--;
            }
        }

        return i + 1; 
    }
}

File: 0027_easy_hot_remove-element/solution.py
from typing import List


class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        if nums is None or len(nums) == 0:
            return 0
        
        n = len(nums)

        i = n - 1
        while i >= 0 and nums[i] == val:
            i -= 1
            
        for j in range(i - 1, -1, -1):
            if nums[j] == val:
                nums[i], nums[j] = nums[j], nums[i]
                i -= 1

        return i + 1

